scripts/download_artifacts.py

```diff
@@ -141,30 +143,38 @@ def artifact_exists(artifact_name: str) -> bool:
     return True
 
 
-def gcc_hashes(git_hash: str, subsequent: bool):
-    """Get the most recent GCC hashes within a 100 commits (in order from closest to furthest)"""
-    if subsequent is False:  # Get prior commit
-        old_commit = (
-            os.popen(
-                f"cd gcc && git checkout master --quiet && git pull --quiet && git rev-parse {git_hash}~100"
-            )
-            .read()
-            .strip()
+def gcc_hashes(git_hash: str, issue_hashes: List[str]):
+    old_commit = (
+        os.popen(
+            f"cd gcc && git checkout master --quiet && git pull --quiet && git rev-parse {git_hash}~10000"
         )
-        print(f"git rev-list --ancestry-path {old_commit}~1..{git_hash}~1")
-        commits = os.popen(
-            f"cd gcc && git rev-list --ancestry-path {old_commit}^..{git_hash}^"
-        ).read()
-        commits = commits.splitlines()
-    else:
-        os.popen("cd gcc && git checkout master --quiet && git pull --quiet").read()
-        commits = os.popen(
-            f"cd gcc && git rev-list --ancestry-path {git_hash}..HEAD | head -100"
-        ).read()
-        commits = list(reversed(commits.splitlines()))
-
-    return commits
-
+        .read()
+        .strip()
+    )
+    print(f"git rev-list --topo-order {old_commit}..{git_hash}")
+    commits = os.popen(
+        f"cd gcc && git rev-list --topo-order {old_commit}..{git_hash}"
+    ).read()
+    commits = commits.splitlines()
+    sorted_issue_hashes = [commit for commit in commits if commit in issue_hashes]
+
+    return sorted_issue_hashes
+
+def issue_hashes(repo_name: str, token: str):
+    params = {
+        "Accept": "application/vnd.github+json",
+        "Authorization": f"token {token}",
+        "X-Github-Api-Version": "2022-11-28",
+    }
+    response = requests.get(
+        f"https://api.github.com/repos/{repo_name}/issues?page=1&per_page=100&state=all",
```

Comment:
I'm guessing the per_page is just a Github thing they put there to show features you wish worked? If this does work we should set it to 400.